enumerables.rb

```diff
@@ -0,0 +1,155 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  # Task-1 my_each exercise
+  def my_each
+    if block_given?
+      length.times { |e| yield(self[e]) }
+    else
+      to_enumerator(:my_each)
+    end
+  end
+
+  # Test for my_each
+  # [1,2,3].my_each {|e| p "#{e}"}
+  # Task-2 my_each_with_index
+  def my_each_with_index
+    if block_given?
+      (0...length).each { |index| yield(self[index], index) }
+    else
+      to_enumerator(:my_each_with_index)
+    end
+  end
+
+  # Test for my_each
+  # [1,2,3].my_each_with_index {|e,i| p "#{e}:#{i}"}
+  # Task-3 my_select
+  def my_select
+    if block_given?
+      arr = []
+      (0...length).each { |element| arr.push(element) if yield(element) }
+      arr
+    else
+      to_enum(:my_select)
+    end
+  end
```

Comment:
#my_select returns an array containing all elements of enum for which the given block returns a true value
#my_select returns an array containing all elements of enum for which the given block returns a true value::range